src/com/facebook/buck/jvm/java/JavaSourceJar.java

```diff
@@ -81,12 +99,21 @@ public JavaSourceJar(
     steps.add(new MakeCleanDirectoryStep(getProjectFilesystem(), temp));
 
     Set<Path> seenPackages = Sets.newHashSet();
+    Map<ProjectFilesystem, JavaPackageFinder> finders = new HashMap<>();
+
+    for (SourcePath sourcePath : sources) {
+      Path source = getResolver().getAbsolutePath(sourcePath);
+      if (!"java".equals(MorePaths.getFileExtension(source))) {
```

Comment:
If you just need to extract the package, a full blown java source code parser might be overkill. Java/Scala/Groovy/Kotlin do not differ in how they declare their package names except for maybe the omission of a semi colon at the end. The package can be extracted via a simpler means (regex?) and the logic can then be applied to all jvm based language source files. Thoughts?